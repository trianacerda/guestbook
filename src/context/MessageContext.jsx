import { useContext, createContext, useState } from 'react';

const MessageContext = createContext();

const MessageProvider = ({ children }) => {
  const [messages, setMessages] = useState('');

  return (
    <MessageContext.Provider value={{ messages, setMessages }}>
      {children}
    </MessageContext.Provider>
  );
};

const useMessages = () => {
  const context = useContext(MessageContext);

  if (context === undefined) {
    throw new Error(
      'useMessages hook must be used withing a MessageContext Provider parent tag'
    );
  }

  return context;
};

export { MessageProvider, useMessages };
