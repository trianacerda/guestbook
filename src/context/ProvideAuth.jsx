import { createContext, useState } from 'react';

export const authContext = createContext();

export function ProvideAuth({ children }) {
  const [user, setUser] = useState(null);

  const login = (email, password) => {
    const loginSuccess =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccess) setUser({ email });
    return loginSuccess;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  return (
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}
