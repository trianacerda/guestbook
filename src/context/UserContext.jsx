import React from 'react';
import { useContext, createContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error(
      'useUser hook must be used withing a MessageContext Provider parent tag'
    );
  }

  return context;
};

export { UserProvider, useUser };
