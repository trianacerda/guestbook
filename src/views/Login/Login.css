label {
  display: flex;
  justify-content: center;
}
form {
  display: flex;
  justify-content: center;
  padding: 30px;
  margin: 30px;
  border-radius: 5px;
  border-width: 10px;
  border-color: aqua;
  background-color: grey;
}
button {
  background-color: gold;
  color: grey;
  border-radius: 20px;
  border-color: aqua;
  border-width: 10px;
  display: flex;
  justify-content: center;
}
