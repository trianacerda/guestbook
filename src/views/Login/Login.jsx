import { useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { useAuth } from '../../hooks/useAuth';
import { useForm } from '../../hooks/useForm';

export default function Login() {
  const history = useHistory();
  const location = useLocation();
  const auth = useAuth();
  const { formState, handleFormStateChange } = useForm({
    email: '',
    password: '',
  });
  const [error, setError] = useState(null);
  const { from } = location.state || { from: { pathname: '/' } };

  const handleLogin = (event) => {
    event.preventDefault();
    const loginWasSuccessful = auth.login(formState.email, formState.password);

    !loginWasSuccessful
      ? setError(
          'whoops--looks like the login information you entered failed. Try again?'
        )
      : history.replace(from);
  };

  return (
    <>
      <form onSubmit={handleLogin}>
        <label aria-label="email">Email</label>
        <input
          id="email"
          name="email"
          type="email"
          value={formState.email}
          onChange={handleFormStateChange}
        />{' '}
        <label aria-label="password">Password</label>
        <input
          id="password"
          name="password"
          type="password"
          value={formState.password}
          onChange={handleFormStateChange}
        />
        <button type="submit" aria-label="sign-in">
          Sign In
        </button>
      </form>
      {error && <h1>{error}</h1>}
    </>
  );
}
