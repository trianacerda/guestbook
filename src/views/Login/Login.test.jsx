import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Login from '../Login/Login';
import { UserProvider } from '../../context/UserContext';
import { ProvideAuth } from '../../context/ProvideAuth';

it('should render Login comp + snapshot test', async () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <ProvideAuth>
          <Login />
        </ProvideAuth>
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
