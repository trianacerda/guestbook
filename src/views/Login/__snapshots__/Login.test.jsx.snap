// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login comp + snapshot test 1`] = `
<div>
  <h1>
     Sign in here to access the guestbook
  </h1>
  <form>
    <label
      aria-label="email"
      style="color: white; font-style: bold; flex-direction: column;"
    >
      Email:
    </label>
    <input
      id="email"
      name="email"
      placeholder="email..."
      required=""
      type="email"
      value=""
    />
     
    <label
      aria-label="password"
      style="color: white; font-style: bold; flex-direction: column;"
    >
      Password:
    </label>
    <input
      id="password"
      name="password"
      placeholder="password..."
      required=""
      type="password"
      value=""
    />
    <button
      aria-label="sign-in"
      type="submit"
    >
      Sign In
    </button>
  </form>
</div>
`;
